internal/build/phases_test.go

```diff
@@ -1,77 +1,36 @@
 package build_test
 
 import (
-	"bytes"
 	"context"
-	"io/ioutil"
 	"math/rand"
-	"os"
-	"path/filepath"
 	"testing"
 	"time"
 
-	"github.com/apex/log"
 	"github.com/docker/docker/api/types/container"
-	"github.com/docker/docker/client"
 	"github.com/heroku/color"
 	"github.com/sclevine/spec"
 	"github.com/sclevine/spec/report"
 
+	"github.com/buildpacks/pack/internal/api"
+
 	"github.com/buildpacks/pack/internal/build"
 	"github.com/buildpacks/pack/internal/build/fakes"
-	ilogging "github.com/buildpacks/pack/internal/logging"
 	h "github.com/buildpacks/pack/testhelpers"
 )
 
-var (
-	phasesRepoName string
-)
-
 func TestPhases(t *testing.T) {
 	rand.Seed(time.Now().UTC().UnixNano())
 
 	color.Disable(true)
 	defer color.Disable(false)
 
-	h.RequireDocker(t)
-
-	dockerCli, err := client.NewClientWithOpts(client.FromEnv, client.WithVersion("1.38"))
-	h.AssertNil(t, err)
-
-	phasesRepoName = "phases.test.lc-" + h.RandString(10)
-
-	wd, err := os.Getwd()
-	h.AssertNil(t, err)
-
-	// Create fake builder
-	h.CreateImageFromDir(t, dockerCli, phasesRepoName, filepath.Join(wd, "testdata", "fake-lifecycle"))
-	defer h.DockerRmi(dockerCli, phasesRepoName)
-
 	spec.Run(t, "phases", testPhases, spec.Report(report.Terminal{}), spec.Sequential())
 }
 
 func testPhases(t *testing.T, when spec.G, it spec.S) {
-	var tmpDir string
-
-	it.Before(func() {
-		var err error
-		tmpDir, err = ioutil.TempDir("", "phases-test")
-		h.AssertNil(t, err)
-
-		dockerConfigDir, err := ioutil.TempDir("", "empty-docker-config-dir")
-		h.AssertNil(t, err)
-
-		h.AssertNil(t, os.Setenv("DOCKER_CONFIG", dockerConfigDir))
```

Comment:
This is necessary to ensure the docker configuration is not contaminated for the tests. For example, if you have logged in to docker, it will have configuration that is not expected in these tests.

Reply:
I believe you but I don't understand - how do these tests even interact with docker?

Reply:
You don't believe me... * nods head *. 😄 

I'm curious myself. Let me follow the crumbs...

1. https://github.com/buildpacks/pack/blob/55b940fbf29d0b4159b2ba02c2964b493722fb5a/internal/build/phases_test.go#L179
2. https://github.com/buildpacks/pack/blob/55b940fbf29d0b4159b2ba02c2964b493722fb5a/internal/build/phases.go#L71
3. https://github.com/buildpacks/pack/blob/55b940fbf29d0b4159b2ba02c2964b493722fb5a/internal/build/phases.go#L91
4. https://github.com/google/go-containerregistry/blob/3d03ed9b1ca2ad5d78d43832e8e46adc31d2b961/pkg/authn/keychain.go#L51
5. https://github.com/google/go-containerregistry/blob/3d03ed9b1ca2ad5d78d43832e8e46adc31d2b961/pkg/authn/keychain.go#L62

As suspected, it's GGCR that does a lookup on the EnvVar based on the keychain provided. We could make that something that we inject via DI.

Reply:
Ohhh okay thank you, this makes sense. I am going to put this back for now, DI sounds like something we want to do but maybe out of scope for this PR.